diff --git a/dist/mermaid.core.mjs b/dist/mermaid.core.mjs
index 13603a2204ed75e59cdd3b59f1dad2a50a0ca1d3..946475ca2dc3522e669f56752b3e5cc118a6a72c 100644
--- a/dist/mermaid.core.mjs
+++ b/dist/mermaid.core.mjs
@@ -3682,8 +3682,9 @@ var render = /* @__PURE__ */ __name(async function(id25, text, svgContainingElem
   const firstChild = svg.firstChild;
   const diagramClassDefs = diag.renderer.getClasses?.(text, diag);
   const rules = createUserStyles(config, diagramType, diagramClassDefs, idSelector);
+  const rulesDark = createUserStyles({...config2, themeVariables: theme.dark.getThemeVariables({})}, diagramType, diagramClassDefs, idSelector);
   const style1 = document.createElement("style");
-  style1.innerHTML = rules;
+  style1.innerHTML = rules + `@media (prefers-color-scheme: dark) {${rulesDark}}`;
   svg.insertBefore(style1, firstChild);
   try {
     await diag.renderer.draw(text, id25, version, diag);
diff --git a/dist/mermaid.esm.mjs b/dist/mermaid.esm.mjs
index 2f09b512ef847144a4bcff16ab39853061f45eb4..bee36b4162e605eaa6a45f7506a402860111da13 100644
--- a/dist/mermaid.esm.mjs
+++ b/dist/mermaid.esm.mjs
@@ -4035,8 +4035,9 @@ var render = /* @__PURE__ */ __name(async function(id25, text, svgContainingElem
   const firstChild = svg.firstChild;
   const diagramClassDefs = diag.renderer.getClasses?.(text, diag);
   const rules = createUserStyles(config, diagramType, diagramClassDefs, idSelector);
+  const rulesDark = createUserStyles({...config2, themeVariables: theme.dark.getThemeVariables({})}, diagramType, diagramClassDefs, idSelector);
   const style1 = document.createElement("style");
-  style1.innerHTML = rules;
+  style1.innerHTML = rules + `@media (prefers-color-scheme: dark) {${rulesDark}}`;
   svg.insertBefore(style1, firstChild);
   try {
     await diag.renderer.draw(text, id25, version, diag);
diff --git a/dist/mermaid.js b/dist/mermaid.js
index 36adb8de5ea722a4c23798b47045faf08a8c7e4f..7f06dd6cfc71bb0b0cb8ac2c62fcc485b709b25e 100644
--- a/dist/mermaid.js
+++ b/dist/mermaid.js
@@ -143251,8 +143251,9 @@ ${config6.themeCSS}`;
     const firstChild = svg.firstChild;
     const diagramClassDefs = diag.renderer.getClasses?.(text2, diag);
     const rules = createUserStyles(config6, diagramType, diagramClassDefs, idSelector);
+    const rulesDark = createUserStyles({...config2, themeVariables: theme.dark.getThemeVariables({})}, diagramType, diagramClassDefs, idSelector);
     const style1 = document.createElement("style");
-    style1.innerHTML = rules;
+    style1.innerHTML = rules + `@media (prefers-color-scheme: dark) {${rulesDark}}`;
     svg.insertBefore(style1, firstChild);
     try {
       await diag.renderer.draw(text2, id27, version2, diag);
