diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 97a639b8bf70a1b8f3fb52bc282991c6c8d31ff6..0000000000000000000000000000000000000000
diff --git a/index.js b/index.js
index b528d85487715360d8fe85ca027fa4800a049281..7c359833209aae782fdc9df244b1b53e7528ec92 100644
--- a/index.js
+++ b/index.js
@@ -35,6 +35,7 @@ var Pjax = function(options) {
     "popstate",
     function(st) {
       if (st.state) {
+        if (st.state.uid === this.lastUid) return;
         var opt = clone(this.options);
         opt.url = st.state.url;
         opt.title = st.state.title;
diff --git a/lib/send-request.js b/lib/send-request.js
index ad7b1b1b633a96b024e3de065b16958a2b4fd107..6ec9e74ee6ad95f08e8c5b390a550cb86aadcf31 100644
--- a/lib/send-request.js
+++ b/lib/send-request.js
@@ -13,11 +13,7 @@ module.exports = function(location, options, callback) {
 
   request.onreadystatechange = function() {
     if (request.readyState === 4) {
-      if (request.status === 200) {
-        callback(request.responseText, request, location, options);
-      } else if (request.status !== 0) {
-        callback(null, request, location, options);
-      }
+      callback(request.responseText, request, location, options);
     }
   };
 
